src/libasr/codegen/asr_to_c.cpp

```diff
@@ -529,7 +524,12 @@ class ASRToCVisitor : public BaseCCPPVisitor<ASRToCVisitor>
             } else if (ASR::is_a<ASR::List_t>(*v_m_type)) {
                 ASR::List_t* t = ASR::down_cast<ASR::List_t>(v_m_type);
                 std::string list_type_c = c_ds_api->get_list_type(t);
-                sub = format_type_c("", list_type_c, v.m_name,
+                std::string name = v.m_name;
+                if (name == "_lpython_return_variable" && v.m_intent == ASRUtils::intent_out) {
```

Comment:
The fix might be as simple as:
```suggestion
                if (v.m_intent == ASRUtils::intent_out) {
```
If it doesn't work, then let's figure out an example where this fails and understand if there are other corner cases to be fixed.

Reply:
Maybe the condition is that if it is a List, it should be passes as a pointer.